Tests/Unit/Controller/FrontEndEditorControllerTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TTN\Tea\Tests\Unit\Controller;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use TTN\Tea\Controller\FrontEndEditorController;
+use TTN\Tea\Domain\Repository\Product\TeaRepository;
+use TYPO3\CMS\Core\Context\Context;
+use TYPO3\CMS\Core\Context\UserAspect;
+use TYPO3\CMS\Core\Http\HtmlResponse;
+use TYPO3\CMS\Core\Information\Typo3Version;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+use TYPO3\CMS\Extbase\Persistence\QueryResultInterface;
+use TYPO3\CMS\Fluid\View\TemplateView;
+use TYPO3\TestingFramework\Core\AccessibleObjectInterface;
+use TYPO3\TestingFramework\Core\Unit\UnitTestCase;
+
+/**
+ * @covers \TTN\Tea\Controller\FrontEndEditorController
+ */
+final class FrontEndEditorControllerTest extends UnitTestCase
+{
+    /**
+     * @var FrontEndEditorController&MockObject&AccessibleObjectInterface
+     */
+    private FrontEndEditorController $subject;
+
+    /**
+     * @var TemplateView&MockObject
+     */
+    private TemplateView $viewMock;
+
+    /**
+     * @var TeaRepository&MockObject
+     */
+    private TeaRepository $teaRepositoryMock;
+
+    private Context $context;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->teaRepositoryMock = $this->createMock(TeaRepository::class);
+        $this->context = new Context();
+
+        // We need to create an accessible mock in order to be able to set the protected `view`.
+        $methodsToMock = ['htmlResponse', 'redirect', 'redirectToUri'];
+        if ((new Typo3Version())->getMajorVersion() <= 11) {
+            $methodsToMock[] = 'forward';
+        }
+        $this->subject = $this->getAccessibleMock(
+            FrontEndEditorController::class,
+            $methodsToMock,
+            [$this->context, $this->teaRepositoryMock]
+        );
+
+        $this->viewMock = $this->createMock(TemplateView::class);
+        $this->subject->_set('view', $this->viewMock);
+
+        $responseMock = $this->createStub(HtmlResponse::class);
```

Comment:
Should we rename `$responseMock` to `$responseStub`, now that it is a stub?

Reply:
Done.